central/telemetry/centralclient/instance_config_test.go

```diff
@@ -13,12 +13,39 @@ import (
 )
 
 func Test_getKey(t *testing.T) {
-	key, err := getKey()
+	cfg, err := getRuntimeConfig()
 	// Telemetry should be disabled in test environment.
-	assert.Empty(t, key)
+	assert.Nil(t, cfg)
 	assert.NoError(t, err)
 }
 
+func Test_agetRuntimeConfig(t *testing.T) {
+	const devVersion = "4.4.1-dev"
+	const remoteKey = "remotekey"
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(`{
+			"storage_key_v1": "` + remoteKey + `",
+			"api_call_campaign": [
+				{"methods": ["put", "delete"]}
+			]
+			}`))
+	}))
+	defer server.Close()
+
+	testutils.SetMainVersion(t, devVersion)
+	t.Setenv(defaults.ImageFlavorEnvName, "opensource")
+	t.Setenv(env.TelemetryConfigURL.EnvVar(), server.URL)
+	t.Setenv(env.TelemetryStorageKey.EnvVar(), remoteKey)
+	cfg, err := getRuntimeConfig()
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+	assert.Equal(t, remoteKey, cfg.Key)
+	require.Len(t, cfg.APICallCampaign, 1)
+	assert.Equal(t, "put", cfg.APICallCampaign[0].Methods[0])
```

Comment:
We can check equality
```suggestion
	require.Equal(t, cfg, config{APICallCampaign: {Methods: [""]})
```